prive/report/report.py

```diff
@@ -0,0 +1,87 @@
+"""Classes to summarise an attacks"""
+from abc import ABC, abstractmethod
+import numpy as np
+
+
+class AttackSummary(ABC):
+
+    @abstractmethod
+    def calculate_metrics(self):
+        """
+        Calculate metrics relevant for an attack
+
+        """
+        pass
+
+    @abstractmethod
+    def write(self, output_path):
+        """
+        Write metrics to file.
+
+        """
+        pass
+
+
+class MiAttackSummary(AttackSummary):
```

Comment:
```suggestion
class MIAttackSummary(AttackSummary):
```
to be consistent with other names